lib/gooddata/lcm/actions/synchronize_user_filters.rb

```diff
@@ -5,6 +5,7 @@
 # LICENSE file in the root directory of this source tree.
```

Comment:
We need to perform a load test which will prove that the parallelism doesn't adversely affect performance and memory consumption. It is important that we don't run into this unresolved issue: https://jira.intgdc.com/browse/MSF-12829

Reply:
I don't have any ideas about how to do it. Could we create a ticket for it and do it later? This PR is checked by @petrvalkoun-gooddata manually.

Reply:
Yes, there is a ticket for automated load tests [here](https://jira.intgdc.com/browse/TMA-605)

For the manual test, we need to ensure that the brick works even with a pessimistic number of clients and new users. I see [from code](https://github.com/gooddata/gooddata-ruby/blob/master/lib/gooddata/models/domain.rb#L259) that it makes `clients.length * users.length` requests all in parallel. This shouldn't be treated lightly.

How many clients and users were in that test? Will it ensure future scalability?

Reply:
actually the number of request is still be controlled by [this](https://github.com/gooddata/gooddata-ruby/blob/develop/lib/gooddata.rb#L8). About the test, could you help me answer it, @petrvalkoun-gooddata ?

Reply:
@panjan  you mean there is nested parallelism? we should do only clients in parallel, not users or filters. The ruby executor has trouble running 20 threads because of CPU bottleneck, running 20*20 would only make it slower and consume more RAM. Until the CPU bottleneck is resolved, we should never run more than 20 threads (which are calling some API) in any ruby brick.

Reply:
The parallelism is nested. First, there's a `peach` [here](https://github.com/vhtien/gooddata-ruby/blob/9d7ce6cf4ecc4198d5d161a9b414b65ae5a90a04/lib/gooddata/lcm/actions/synchronize_user_filters.rb#L189). Second, there's a `pmapcat` [here](https://github.com/gooddata/gooddata-ruby/blob/develop/lib/gooddata.rb#L8).

I agree with @petrvalkoun-gooddata that we should not nest `peach`es as it would fire up 400 threads with default settings.

We must definitely test this with bigger numbers. Say 2K clients, 1M users.

Reply:
well, I think we need to use another gem. We cannot get rid of nested parallelism because it will slow down other single tasks. I mentioned this issue before (https://jira.intgdc.com/browse/TMA-416)

Reply:
We'll need to resolve TMA-416 first then.